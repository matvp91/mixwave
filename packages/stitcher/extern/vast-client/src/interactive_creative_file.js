import { parserUtils } from "./parser/parser_utils.js";

export function createInteractiveCreativeFile(
  interactiveCreativeAttributes = {},
) {
  return {
    type: interactiveCreativeAttributes.type || null,
    apiFramework: interactiveCreativeAttributes.apiFramework || null,
    variableDuration: parserUtils.parseBoolean(
      interactiveCreativeAttributes.variableDuration,
    ),
    fileURL: null,
  };
}
