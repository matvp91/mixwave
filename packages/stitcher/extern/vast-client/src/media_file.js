export function createMediaFile() {
  return {
    id: null,
    fileURL: null,
    fileSize: 0,
    deliveryType: 'progressive',
    mimeType: null,
    mediaType: null,
    codec: null,
    bitrate: 0,
    minBitrate: 0,
    maxBitrate: 0,
    width: 0,
    height: 0,
    apiFramework: null, // @deprecated in VAST 4.1. <InteractiveCreativeFile> should be used instead.
    scalable: null,
    maintainAspectRatio: null,
  };
}
