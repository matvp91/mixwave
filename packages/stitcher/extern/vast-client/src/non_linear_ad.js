export function createNonLinearAd() {
  return {
    id: null,
    width: 0,
    height: 0,
    expandedWidth: 0,
    expandedHeight: 0,
    scalable: true,
    maintainAspectRatio: true,
    minSuggestedDuration: 0,
    apiFramework: 'static',
    adType: 'nonLinearAd',
    type: null,
    staticResource: null,
    htmlResource: null,
    iframeResource: null,
    nonlinearClickThroughURLTemplate: null,
    nonlinearClickTrackingURLTemplates: [],
    adParameters: null,
  };
}

export function isNonLinearAd(ad) {
  return ad.adType === 'nonLinearAd';
}
