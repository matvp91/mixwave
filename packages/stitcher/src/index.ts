import Fastify from "fastify";
import cors from "@fastify/cors";
import { env } from "./env.js";
import { packageBaseUrl } from "./const.js";
import { randomUUID } from "crypto";
import { masterPlaylist, mediaPlaylist } from "./playlist.js";
import pnc from "persistent-node-cache";

const cache = new pnc.PersistentNodeCache("stitcher");

async function buildServer() {
  const app = Fastify();

  app.register(cors);

  app.get<{
    Params: {
      sessionId: string;
    };
  }>("/hls/:sessionId/master.m3u8", async (request, reply) => {
    const session = cache.get<{ assetId: string }>(request.params.sessionId);
    if (!session) {
      throw new Error("missing session");
    }

    const response = await masterPlaylist(
      `${packageBaseUrl}/${session.assetId}/hls/master.m3u8`,
    );

    reply.type("application/x-mpegURL").send(response);
  });

  app.get<{
    Params: {
      sessionId: string;
      path: string;
    };
  }>("/hls/:sessionId/:path/playlist.m3u8", async (request, reply) => {
    const session = cache.get<{ assetId: string }>(request.params.sessionId);
    if (!session) {
      throw new Error("missing session");
    }

    const response = await mediaPlaylist(
      `${packageBaseUrl}/${session.assetId}/hls/${request.params.path}/playlist.m3u8`,
    );

    reply.type("application/x-mpegURL").send(response);
  });

  app.get("/", (request, reply) => {
    const assetId = "010d2293-e793-480a-9504-faf68d5ef0df";
    const sessionId = randomUUID();

    cache.set(
      sessionId,
      {
        assetId,
      },
      1000 * 60 * 60 * 24,
    );

    reply.send({
      url: `${env.HOST}/hls/${sessionId}/master.m3u8`,
    });
  });

  return app;
}

async function main() {
  const app = await buildServer();

  await app.listen({ port: env.PORT });
}

main();
