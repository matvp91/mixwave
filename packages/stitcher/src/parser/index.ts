export { parseMasterPlaylist, parseMediaPlaylist } from "./parse";
export { stringifyMasterPlaylist, stringifyMediaPlaylist } from "./stringify";

export * from "./types";
