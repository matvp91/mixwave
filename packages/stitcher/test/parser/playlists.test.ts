import "bun";
import { test, expect, describe, beforeEach, setSystemTime } from "bun:test";
import * as fs from "node:fs";
import { parseMasterPlaylist, parseMediaPlaylist } from "../../src/parser";

async function readPlaylistFixtures() {
  const path = `${__dirname}/fixtures/playlists`;
  const files = fs.readdirSync(path);

  return Promise.all(
    files.map(async (file) => {
      return [file, await Bun.file(`${path}/${file}`).text()];
    }),
  );
}

describe("playlists", async () => {
  beforeEach(() => {
    // The day my son was born!
    setSystemTime(new Date(2021, 4, 2, 10, 12, 5, 250));
  });

  test.each(await readPlaylistFixtures())("playlist(%s)", (name, text) => {
    if (name.startsWith("master-")) {
      expect(parseMasterPlaylist(text)).toMatchSnapshot();
    }
    if (name.startsWith("media-")) {
      expect(parseMediaPlaylist(text)).toMatchSnapshot();
    }
  });
});
