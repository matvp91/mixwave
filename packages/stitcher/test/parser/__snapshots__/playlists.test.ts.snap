// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`playlists > playlist(master-audioChannels.m3u8) 1`] = `
{
  "independentSegments": false,
  "variants": [
    {
      "audio": [
        {
          "channels": "6",
          "groupId": "audio-hi",
          "name": "Espanol Surround",
          "type": "AUDIO",
          "uri": "sp/prog_index1.m3u8",
        },
        {
          "channels": "2",
          "groupId": "audio-hi",
          "name": "Espanol Stereo",
          "type": "AUDIO",
          "uri": "sp/prog_index2.m3u8",
        },
      ],
      "bandwidth": 195023,
      "codecs": "mp4a.40.5",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo/prog_index.m3u8",
    },
    {
      "audio": [],
      "bandwidth": 195023,
      "codecs": "mp4a.40.5",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo/prog_index.m3u8",
    },
  ],
}
`;

exports[`playlists > playlist(master-fmp4.m3u8) 1`] = `
{
  "independentSegments": true,
  "variants": [
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 2215219,
      "codecs": "avc1.640020,mp4a.40.2",
      "resolution": {
        "height": 540,
        "width": 960,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v4/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 7976430,
      "codecs": "avc1.64002a,mp4a.40.2",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v8/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 6181885,
      "codecs": "avc1.64002a,mp4a.40.2",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v7/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 4682666,
      "codecs": "avc1.64002a,mp4a.40.2",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v6/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 3170746,
      "codecs": "avc1.640020,mp4a.40.2",
      "resolution": {
        "height": 720,
        "width": 1280,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v5/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 1276223,
      "codecs": "avc1.64001e,mp4a.40.2",
      "resolution": {
        "height": 432,
        "width": 768,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v3/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 904744,
      "codecs": "avc1.64001e,mp4a.40.2",
      "resolution": {
        "height": 360,
        "width": 640,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v2/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud1",
          "name": "English",
          "type": "AUDIO",
          "uri": "a1/prog_index.m3u8",
        },
      ],
      "bandwidth": 538201,
      "codecs": "avc1.640015,mp4a.40.2",
      "resolution": {
        "height": 270,
        "width": 480,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v1/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 2440329,
      "codecs": "avc1.640020,ac-3",
      "resolution": {
        "height": 540,
        "width": 960,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v4/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 8201540,
      "codecs": "avc1.64002a,ac-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v8/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 6406995,
      "codecs": "avc1.64002a,ac-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v7/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 4907776,
      "codecs": "avc1.64002a,ac-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v6/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 3395856,
      "codecs": "avc1.640020,ac-3",
      "resolution": {
        "height": 720,
        "width": 1280,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v5/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 1501333,
      "codecs": "avc1.64001e,ac-3",
      "resolution": {
        "height": 432,
        "width": 768,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v3/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 1129854,
      "codecs": "avc1.64001e,ac-3",
      "resolution": {
        "height": 360,
        "width": 640,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v2/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud2",
          "name": "English",
          "type": "AUDIO",
          "uri": "a2/prog_index.m3u8",
        },
      ],
      "bandwidth": 763311,
      "codecs": "avc1.640015,ac-3",
      "resolution": {
        "height": 270,
        "width": 480,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v1/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 2248329,
      "codecs": "avc1.640020,ec-3",
      "resolution": {
        "height": 540,
        "width": 960,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v4/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 8009540,
      "codecs": "avc1.64002a,ec-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v8/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 6214995,
      "codecs": "avc1.64002a,ec-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v7/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 4715776,
      "codecs": "avc1.64002a,ec-3",
      "resolution": {
        "height": 1080,
        "width": 1920,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v6/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 3203856,
      "codecs": "avc1.640020,ec-3",
      "resolution": {
        "height": 720,
        "width": 1280,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v5/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 1309333,
      "codecs": "avc1.64001e,ec-3",
      "resolution": {
        "height": 432,
        "width": 768,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v3/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 937854,
      "codecs": "avc1.64001e,ec-3",
      "resolution": {
        "height": 360,
        "width": 640,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v2/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "aud3",
          "name": "English",
          "type": "AUDIO",
          "uri": "a3/prog_index.m3u8",
        },
      ],
      "bandwidth": 571311,
      "codecs": "avc1.640015,ec-3",
      "resolution": {
        "height": 270,
        "width": 480,
      },
      "subtitles": [
        {
          "channels": undefined,
          "groupId": "sub1",
          "name": "English",
          "type": "SUBTITLES",
          "uri": "s1/eng/prog_index.m3u8",
        },
      ],
      "uri": "v1/prog_index.m3u8",
    },
  ],
}
`;

exports[`playlists > playlist(master-multipleAudioGroups.m3u8) 1`] = `
{
  "independentSegments": false,
  "variants": [
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "English",
          "type": "AUDIO",
          "uri": "englo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Français",
          "type": "AUDIO",
          "uri": "frelo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "splo/prog_index.m3u8",
        },
      ],
      "bandwidth": 195023,
      "codecs": "mp4a.40.5",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "English",
          "type": "AUDIO",
          "uri": "englo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Français",
          "type": "AUDIO",
          "uri": "frelo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "splo/prog_index.m3u8",
        },
      ],
      "bandwidth": 260000,
      "codecs": "avc1.42e01e,mp4a.40.2",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo2/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "English",
          "type": "AUDIO",
          "uri": "eng/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Français",
          "type": "AUDIO",
          "uri": "fre/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "sp/prog_index.m3u8",
        },
      ],
      "bandwidth": 591680,
      "codecs": "mp4a.40.2, avc1.64001e",
      "resolution": undefined,
      "subtitles": [],
      "uri": "hi/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "English",
          "type": "AUDIO",
          "uri": "eng/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Français",
          "type": "AUDIO",
          "uri": "fre/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "sp/prog_index.m3u8",
        },
      ],
      "bandwidth": 650000,
      "codecs": "avc1.42e01e,mp4a.40.2",
      "resolution": undefined,
      "subtitles": [],
      "uri": "hi2/prog_index.m3u8",
    },
  ],
}
`;

exports[`playlists > playlist(master-multipleAudioGroupsCombined.m3u8) 1`] = `
{
  "independentSegments": false,
  "variants": [
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "English",
          "type": "AUDIO",
          "uri": undefined,
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Français",
          "type": "AUDIO",
          "uri": "frelo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "splo/prog_index.m3u8",
        },
      ],
      "bandwidth": 195023,
      "codecs": "mp4a.40.5",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "English",
          "type": "AUDIO",
          "uri": undefined,
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Français",
          "type": "AUDIO",
          "uri": "frelo/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-lo",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "splo/prog_index.m3u8",
        },
      ],
      "bandwidth": 260000,
      "codecs": "avc1.42e01e,mp4a.40.2",
      "resolution": undefined,
      "subtitles": [],
      "uri": "lo2/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "English",
          "type": "AUDIO",
          "uri": "eng/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Français",
          "type": "AUDIO",
          "uri": "fre/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "sp/prog_index.m3u8",
        },
      ],
      "bandwidth": 591680,
      "codecs": "mp4a.40.2, avc1.64001e",
      "resolution": undefined,
      "subtitles": [],
      "uri": "hi/prog_index.m3u8",
    },
    {
      "audio": [
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "English",
          "type": "AUDIO",
          "uri": "eng/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Français",
          "type": "AUDIO",
          "uri": "fre/prog_index.m3u8",
        },
        {
          "channels": undefined,
          "groupId": "audio-hi",
          "name": "Espanol",
          "type": "AUDIO",
          "uri": "sp/prog_index.m3u8",
        },
      ],
      "bandwidth": 650000,
      "codecs": "avc1.42e01e,mp4a.40.2",
      "resolution": undefined,
      "subtitles": [],
      "uri": "hi2/prog_index.m3u8",
    },
  ],
}
`;

exports[`playlists > playlist(master-simple.m3u8) 1`] = `
{
  "independentSegments": false,
  "variants": [
    {
      "audio": [],
      "bandwidth": 240000,
      "codecs": undefined,
      "resolution": {
        "height": 224,
        "width": 396,
      },
      "subtitles": [],
      "uri": "media.m3u8",
    },
    {
      "audio": [],
      "bandwidth": 40000,
      "codecs": undefined,
      "resolution": undefined,
      "subtitles": [],
      "uri": "media1.m3u8",
    },
    {
      "audio": [],
      "bandwidth": 440000,
      "codecs": undefined,
      "resolution": {
        "height": 224,
        "width": 396,
      },
      "subtitles": [],
      "uri": "media2.m3u8",
    },
    {
      "audio": [],
      "bandwidth": 1928000,
      "codecs": undefined,
      "resolution": {
        "height": 540,
        "width": 960,
      },
      "subtitles": [],
      "uri": "media3.m3u8",
    },
  ],
}
`;

exports[`playlists > playlist(media-dateTime.m3u8) 1`] = `
{
  "dateRanges": [],
  "discontinuitySequenceBase": undefined,
  "endlist": true,
  "independentSegments": false,
  "mediaSequenceBase": 0,
  "playlistType": "VOD",
  "segments": [
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "hls_450k_video.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": "2016-06-22T13:20:26.166+00:00",
      "uri": "hls_450k_video.ts",
    },
  ],
  "targetDuration": 10,
}
`;

exports[`playlists > playlist(media-fmp4.m3u8) 1`] = `
{
  "dateRanges": [],
  "discontinuitySequenceBase": undefined,
  "endlist": true,
  "independentSegments": true,
  "mediaSequenceBase": 1,
  "playlistType": "VOD",
  "segments": [
    {
      "discontinuity": undefined,
      "duration": 6.006,
      "map": {
        "uri": "main.mp4",
      },
      "programDateTime": undefined,
      "uri": "main.mp4",
    },
    {
      "discontinuity": undefined,
      "duration": 6.006,
      "map": {
        "uri": "main.mp4",
      },
      "programDateTime": undefined,
      "uri": "main.mp4",
    },
  ],
  "targetDuration": 6,
}
`;

exports[`playlists > playlist(media-interstitials.m3u8) 1`] = `
{
  "dateRanges": [
    {
      "classId": "com.apple.hls.interstitial",
      "clientAttributes": {
        "ASSET-LIST": "http://localhost:52002/session/b41fa084-fad0-47f7-af9e-7f03c2e81edb/asset-list.json?startDate",
        "RESTRICT": "SKIP,JUMP",
        "RESUME-OFFSET": 0,
      },
      "id": "i0",
      "startDate": "2024-10-07T17:17:00.207+00:00",
    },
    {
      "classId": "com.apple.hls.interstitial",
      "clientAttributes": {
        "ASSET-LIST": "http://localhost:52002/session/b41fa084-fad0-47f7-af9e-7f03c2e81edb/asset-list.json?startDate",
        "RESTRICT": "SKIP,JUMP",
        "RESUME-OFFSET": 0,
      },
      "id": "i1",
      "startDate": "2024-10-07T17:17:10.207+00:00",
    },
  ],
  "discontinuitySequenceBase": undefined,
  "endlist": true,
  "independentSegments": false,
  "mediaSequenceBase": undefined,
  "playlistType": "VOD",
  "segments": [
    {
      "discontinuity": undefined,
      "duration": 4,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "1.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 4,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "2.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 4,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "3.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 4,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "4.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 4,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "5.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 5.833,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "6.m4s",
    },
    {
      "discontinuity": undefined,
      "duration": 3.792,
      "map": {
        "uri": "init.mp4",
      },
      "programDateTime": undefined,
      "uri": "7.m4s",
    },
  ],
  "targetDuration": 6,
}
`;

exports[`playlists > playlist(media-mediaSequence.m3u8) 1`] = `
{
  "dateRanges": [],
  "discontinuitySequenceBase": undefined,
  "endlist": true,
  "independentSegments": false,
  "mediaSequenceBase": 0,
  "playlistType": "VOD",
  "segments": [
    {
      "discontinuity": undefined,
      "duration": 6.64,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "/test/ts-files/tvy7/8a5e2822668b5370f4eb1438b2564fb7ab12ffe1-hi720.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 6.08,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "/test/ts-files/tvy7/56be1cef869a1c0cc8e38864ad1add17d187f051-hi720.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 6.6,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "/test/ts-files/tvy7/549c8c77f55f049741a06596e5c1e01dacaa46d0-hi720.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 5,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "/test/ts-files/tvy7/6cfa378684ffeb1c455a64dae6c103290a1f53d4-hi720.ts",
    },
  ],
  "targetDuration": 8,
}
`;

exports[`playlists > playlist(media-simple.m3u8) 1`] = `
{
  "dateRanges": [],
  "discontinuitySequenceBase": undefined,
  "endlist": true,
  "independentSegments": false,
  "mediaSequenceBase": undefined,
  "playlistType": "VOD",
  "segments": [
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "media-00001.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "media-00002.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "media-00003.ts",
    },
    {
      "discontinuity": undefined,
      "duration": 10,
      "map": undefined,
      "programDateTime": undefined,
      "uri": "media-00004.ts",
    },
  ],
  "targetDuration": 10,
}
`;
